### PR TITLE
#1033 Migrating GEOADD and GEODIST command

### PR DESCRIPTION
**Pull request by YuvrajGosain**
Issue: #1033 

 - [x] Migrated the evalXXX function with the latest definition
 - [x] Update or add unit tests for the new implementation.
 - [x] All unit tests pass successfully.
 - [x] Ensure all integration tests pass successfully.
 - [x] Ensure integration tests are added for the migrated command on multi-threaded resp server.
 - [x] Move Integration tests for the respective commands under the RESP integration tests directory from Async directory
 - [x] Please validate that the documentation for the respective commands is up to date. If not then consider adding them.

### CODE REVIEW

**Review by lucifercr07 — CHANGES_REQUESTED** [review]
@YuvrajGosain can follow sample doc [template](https://github.com/DiceDB/dice/blob/c3400e88c8f647392840f1caa83c69c635c1bee7/docs/sample_command_docs.md) and update for these commands accordingly it'd be helpful.

**Comment by lucifercr07** [comment]
@YuvrajGosain please resolve conflicts once.

**Comment by YuvrajGosain** [comment]
Hi @AshwinKul28 , @lucifercr07 , Apologies for the late reply , was busy in some personal stuff and wasn't able to check the status for this, I'll rebase the PR and add docs for these commands by today.

**Comment by YuvrajGosain** [comment]
Hi @AshwinKul28 , @lucifercr07 , I have rebased this branch with the master and added the docs for both the commands as well.

File: docs/src/content/docs/commands/GEOADD.md
Comment (helios2003) [inline docs/src/content/docs/commands/GEOADD.md:1]: Hi, @YuvrajGosain, there are typos in the spellings of longitude and latitude, found them in Line 54 and Line 58  :)

Comment (YuvrajGosain) [reply docs/src/content/docs/commands/GEOADD.md:1]: Hi @helios2003 , this should be resolved now :)

File: docs/src/content/docs/commands/GEOADD.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+3. `Latitude not a valid number or is out of range `
+   - Error Message: (error) ERR invalid latitude.
+   - Occurs when latitude is out of range(-85.05112878 to 85.05112878) or not a valid number.
+
+## Example Usage
+
+Here are a few examples demonstrating the usage of the GEOADD command:
+
+### Example 1: Adding new member to a set
```
Comment (c-harish) [inline docs/src/content/docs/commands/GEOADD.md:66]: Drop the numbering of examples as 1, 2, etc. You can refer to the sample doc [here](https://github.com/DiceDB/dice/blob/master/docs/command_docs_template.md).

File: docs/src/content/docs/commands/GEODIST.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+1.`Wrong number of arguments for 'GEODIST' command`
+   - Error Message: (error) ERR wrong number of arguments for 'geodist' command.
+   - Occurs when the command is executed with an incorrect number of arguments.
+
+
+## Example Usage
+
+Here are a few examples demonstrating the usage of the GEODIST command:
+
+### Example 1: Adding new member to a set
```
Comment (c-harish) [inline docs/src/content/docs/commands/GEODIST.md:58]: Same here.

File: integration_tests/commands/http/geo_tests.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package http
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
```
Comment (c-harish) [inline integration_tests/commands/http/geo_tests.go:6]: Please shift to testify for assertions. You can refer to other test files for example.

File: integration_tests/commands/resp/geo_tests.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
+package resp
+
+import (
+	"testing"
+	"time"
+
+	"gotest.tools/v3/assert"
```
Comment (c-harish) [inline integration_tests/commands/resp/geo_tests.go:7]: Please shift to testify for assertions. You can refer to other test files for example.

File: integration_tests/commands/websocket/geo_tests.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
+package websocket
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Comment (c-harish) [inline integration_tests/commands/websocket/geo_tests.go:6]: Please shift to testify for assertions. You can refer to other test files for example.

**Review by c-harish — CHANGES_REQUESTED** [review]

**Comment by JyotinderSingh** [comment]
@YuvrajGosain Please address the remaining review comments

File: integration_tests/commands/resp/geo_tests.go (hunk 4 above)
Comment (YuvrajGosain) [reply integration_tests/commands/resp/geo_tests.go]: Hi @c-harish , I have addressed this comment.

File: integration_tests/commands/http/geo_tests.go (hunk 3 above)
Comment (YuvrajGosain) [reply integration_tests/commands/http/geo_tests.go]: Hi @c-harish , I have addressed this comment.

File: docs/src/content/docs/commands/GEODIST.md (hunk 2 above)
Comment (YuvrajGosain) [reply docs/src/content/docs/commands/GEODIST.md]: Hi @c-harish , I have addressed this comment.

File: docs/src/content/docs/commands/GEOADD.md (hunk 1 above)
Comment (YuvrajGosain) [reply docs/src/content/docs/commands/GEOADD.md]: Hi @c-harish , I have addressed this comment.

**Comment by YuvrajGosain** [comment]
> @YuvrajGosain Please address the remaining review comments

@JyotinderSingh Hi , I have addressed the comments and rebased the branch with master. Pls have a look

File: docs/src/content/docs/commands/GEOADD.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+| Incorrect Argument Count                                     |`ERR wrong number of arguments for 'geoadd' command`         |
+| If the longitude is not a valid number or is out of range.   |`ERR invalid longitude`                                      |
+| If the latitude  is not a valid number or is out of range.   |`ERR invalid latitude`                                       |
+
+## Behaviour
+
+When the GEOADD command is issued, DiceDB performs the following steps:
+
+1. It checks if argument count is valid or not. If not an error is thrown.
+2. It checks whether optional parameters are applied or not.
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/GEOADD.md:41]: These points are not required.

File: docs/src/content/docs/commands/GEODIST.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+---
+title: GEODIST
+description: The `GEODIST` command in Redis is used to calculate the distance between two members (geospatial points) stored in a geospatial index(set). 
+---
+
+The `GEODIST` command in Redis is used to calculate the distance between two members (geospatial points) stored in a geospatial index(set).
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/GEODIST.md:6]:
```suggestion
The `GEODIST` command in DiceDB is used to calculate the distance between two members (geospatial points) stored in a geospatial index(set).
```

File: docs/src/content/docs/commands/GEODIST.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+| If both members exist in the set with option ft              | distance b/w them in feet                                   |
+| If both members exist in the set with option mi              | distance b/w them in miles                                  |
+| If any member doesn't exist in Set                           | nil                                                         |
+| Incorrect Argument Count                                     |`ERR wrong number of arguments for 'geodist' command`        |
+
+## Behaviour
+
+When the GEODIST command is issued, DiceDB performs the following steps:
+
+1. It checks if argument count is valid or not. If not an error is thrown.
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/GEODIST.md:42]: this point is not needed.

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for this massive migration effort @YuvrajGosain! The changes mostly look good, just had a few minor comments.

File: docs/src/content/docs/commands/GEOADD.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+3. It checks whether longitude and latitude are valid or not. If not an error is thrown. 
+3. It checks whether the set exists or not.
+4. If set doesn't exist new set is created or else the same set is used. 
+5. It adds or updates the member in the set.
+6. It returns number of members added.
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/GEOADD.md:46]:
```suggestion
1. It checks whether longitude and latitude are valid or not. If not an error is thrown. 
2. It checks whether the set exists or not.
3. If set doesn't exist new set is created or else the same set is used. 
4. It adds or updates the member in the set.
5. It returns number of members added. 
```

File: docs/src/content/docs/commands/GEODIST.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+2. It gets the sorted set(key).
+3. It gets the scores(geohashes) from the sorted sets for both the members.
+4. It calculates the distance bw them and returns it.
```
Comment (JyotinderSingh) [inline docs/src/content/docs/commands/GEODIST.md:45]:
```suggestion
1. It gets the sorted set(key).
2. It gets the scores(geohashes) from the sorted sets for both the members.
3. It calculates the distance bw them and returns it.
```

**Comment by YuvrajGosain** [comment]
Hi @JyotinderSingh , just realized that I also pushed the changes onto the branch simultaneously and saw it being merged :)

**Pull request merged**